### PR TITLE
Multiple buckets support

### PR DESCRIPTION
**Pull request by nl0**
### Features:
- multiple buckets support
  - per-bucket search, packages, file browser and overview
  - nav bar bucket controls

### Infrastructure:

- update Material-UI
- `utils/NamedRoutes` module for injecting route configuration
- refactor layout
- refactor `Footer` component
- remove obsolete components
- adjust config format, hardcode buckets
- `utils/ErrorBoundary` module
- use error boundary for 404s
- `utils/Data` module for declarative data fetching
- split notifications into provider / display
- factor-out search provider
- `utils/sagaTools`: helpers for working with `tagged`

### CODE REVIEW

**Comment by nl0** [comment]
This PR is getting huge, so maybe we should merge it while it's more or less stable.

File: catalog/app/containers/NavBar/BucketControls.js [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+                  secondaryTypographyProps={{
+                    noWrap: true,
+                    className: classes.description,
+                  }}
+                  title={s.description}
+                />
+              </MenuItem>
+            ))}
+            <li className={classes.help}>
+              <a href="TODO">Learn how to create your own registry</a>
```
Comment (nl0) [inline catalog/app/containers/NavBar/BucketControls.js]: @ResidentMario where to?

File: catalog/app/containers/Bucket/errors.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      >
+        Learn how to configure the bucket for T4
+      </a>.
+    </Message>
+  )],
+  [R.is(AccessDenied), () => (
+    <Message headline="Access Denied">
+      Seems like you don`t have access to this bucket.
+      <br />
+      <a href="TODO">Learn about access control in T4</a>.
```
Comment (nl0) [inline catalog/app/containers/Bucket/errors.js]: @ResidentMario where to?

Comment (ResidentMario) [reply catalog/app/containers/Bucket/errors.js]: https://github.com/quiltdata/t4/tree/master/deployment#permissions

File: catalog/app/containers/NavBar/BucketControls.js (hunk 1 above)
Comment (ResidentMario) [reply catalog/app/containers/NavBar/BucketControls.js]: https://github.com/quiltdata/t4/tree/master/deployment#installation

File: catalog/app/containers/Bucket/errors.js (hunk 2 above)
Comment (nl0) [reply catalog/app/containers/Bucket/errors.js]:
```suggestion
      <a href="https://github.com/quiltdata/t4/tree/master/deployment#permissions">
        Learn about access control in T4
      </a>.
```

File: catalog/app/containers/NavBar/BucketControls.js (hunk 1 above)
Comment (nl0) [reply catalog/app/containers/NavBar/BucketControls.js]:
```suggestion
              <a href="https://github.com/quiltdata/t4/tree/master/deployment#installation">
                Learn how to create your own registry
              </a>
```

**Pull request merged**